c2rust-analyze/src/c_void_casts.rs

```diff
@@ -274,16 +287,88 @@ impl<'tcx> CVoidCasts<'tcx> {
     /// [`Cast`]: rustc_middle::mir::Rvalue::Cast
     /// [`From`]: CVoidCastDirection::From
     /// [`To`]: CVoidCastDirection::To
-    pub fn should_skip_stmt(&self, stmt: &Statement<'tcx>) -> bool {
-        let (lhs, rv) = match get_assign_sides(stmt) {
-            None => return false,
-            Some(it) => it,
-        };
-        self.to.contains(lhs)
-            || self.from.contains(match get_cast_place(rv) {
-                None => return false,
-                Some(it) => it,
-            })
+    pub fn should_skip_stmt(&self, loc: &Location) -> bool {
+        self.to.contains(loc) || self.from.contains(loc)
+    }
+
+    fn is_place_local_modified(p: &Place, stmt: &Statement) -> bool {
+        if !p.projection.is_empty() {
+            return false;
+        }
```

Comment:
What is this function checking, exactly?  I'm not sure whether this first case should return `false` because `p` isn't a local, or `true` because we can't be sure that `stmt` leaves `p` unmodified.

Reply:
In particular it was meant to check what your [comment](https://github.com/immunant/c2rust/pull/788#discussion_r1113481984) outlined:

> I think we can conservatively treat the Place as unmodified only if (1) it's just a Local, with no projections, and (2) that local isn't mentioned in the LHS or RHS of any statements between the cast and the call.

Reply:
Can you add that as docs?

Reply:
93a30394